backend/controllers/image.js

```diff
@@ -0,0 +1,78 @@
+    description: req.body.image.description, // set the description property of the Image object to the description property of the request body
+    ImageUrl: url + '/Images/' + req.file.filename, // set the ImageUrl property of the Image object to the ImageUrl property of the request body
+    price: req.body.image.price, // set the price property of the Image object to the price property of the request body
+    userId: req.body.image.userId, // set the userId property of the Image object to the userId property of the request body
+  }); // end of const image = new Image({ ... })
+  image.save() // call the save method, which saves the Image object to the database
+    .then(() => res.status(201).json({ message: 'Image saved successfully!' })) // call the then method, which adds a callback function to the promise, to handle the success case
+    .catch(error => res.status(400).json({ error: error })); // call the catch method, which adds a callback function to the promise, to handle the failure case
+}
+
+exports.getOneImage = (req, res, next) => { // call the get method, which adds a route to the router object, to handle GET requests to the /:id endpoint
+  Image.findOne({ _id: req.params.id }) // call the findOne method, which returns a promise, which resolves to the Image object with the specified id
+    .then(image => res.status(200).json(image)) // call the then method, which adds a callback function to the promise, to handle the success case
+    .catch(error => res.status(404).json({ error: error })); // call the catch method, which adds a callback function to the promise, to handle the failure case
+}
+
+exports.modifyImage = (req, res, next) => { // call the put method, which adds a route to the router object, to handle PUT requests to the /:id endpoint
+  let image = new Image({ _id: req.params._id }); // create a new Image object, using the Image model
+  if (req.file) { // if the request contains a file
+    const url = req.protocol + '://' + req.get('host'); // set the url constant to the protocol and host of the request
+    req.body.image = JSON.parse(req.body.image); // parse the request body
+    image = { // set the image object to an object with the following properties
+      _id: req.params.id, // set the _id property of the Image object to the id property of the request parameters
+      title: req.body.image.title, // set the title property of the Image object to the title property of the request body
+      description: req.body.image.description, // set the description property of the Image object to the description property of the request body
+      ImageUrl: url + '/Images/' + req.file.filename, // set the ImageUrl property of the Image object to the ImageUrl property of the request body
+      price: req.body.image.price, // set the price property of the Image object to the price property of the request body
```

Comment:
Same as last comment. What fields are needed for images? This is likely to carry for all the Image functions here. Sorry if I am just missing something.